app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModel.kt

```diff
@@ -111,15 +112,18 @@ class NewConversationViewModel
     }
 
     private suspend fun searchKnown(searchTerm: String) {
-        // TODO: this is going to be refactored on the Kalium side so that we do not use Flow
-        searchKnownUsers(searchTerm).onStart {
-            localContactSearchResult = ContactSearchResult.InternalContact(SearchResultState.InProgress)
-        }.catch {
-            localContactSearchResult = ContactSearchResult.InternalContact(SearchResultState.Failure())
-        }.flowOn(Dispatchers.IO).collect {
-            localContactSearchResult = ContactSearchResult.InternalContact(
-                SearchResultState.Success(it.result.map { otherUser -> otherUser.toContact() })
-            )
+        localContactSearchResult = ContactSearchResult.InternalContact(SearchResultState.InProgress)
+
+        viewModelScope.launch {
```

Comment:
do we need to launch this logic on a separate thread if even the function is defined as suspendable?